src/dataset.hpp

```diff
@@ -62,7 +62,13 @@ struct randomx_cache {
 //A pointer to a standard-layout struct object points to its initial member
 static_assert(std::is_standard_layout<randomx_dataset>(), "randomx_dataset must be a standard-layout struct");
 //the following assert fails when compiling Debug in Visual Studio (JIT mode will crash in Debug)
+#if _MSC_VER && !__INTEL_COMPILER && _DEBUG
```

Comment:
This should be `#if defined(_MSC_VER) && !defined(__INTEL_COMPILER) && defined(_DEBUG)`

Reply:
Done